addon/components/file-browser-icon/component.js

```diff
@@ -0,0 +1,31 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+  layout,
+  tagName: 'span',
+  classNames: ['file-browser-icon'],
+
+  click() {
+      if (this.get('row.isExpandable')) {
+          this.toggleProperty('row.expanded');
+      }
+  },
+
+  iconName: Ember.computed('row', 'row.expanded', function() {
```

Comment:
`classNameBindings` would be better if the component's element were the icon itself, but it's a container that could hold multiple icons (like the expandy caret for folders).